src/profile-logic/marker-data.js

```diff
@@ -340,7 +343,7 @@ export function deriveMarkersFromRawMarkerTable(
   }
 
   matchedMarkers.sort((a, b) => a.start - b.start);
-  return matchedMarkers;
+  return matchedMarkers.map((marker, i) => ({ ...marker, markerIndex: i }));
```

Comment:
I don't like looping once more here, but I thought it was the simplest solution.
The more complex solution involves using arrays of marker indexes for the filtetered arrays, instead of using arrays of Markers like today.

Reply:
The new code mutates the object.